### PR TITLE
[Instrumentation.Process] Update .NET API for virtual memory usage

### PR DESCRIPTION
**Pull request by Yun-Ting**
Addressing https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/664#discussion_r993386421

Spec issue:
https://github.com/open-telemetry/opentelemetry-specification/issues/2928

### CODE REVIEW

**Comment by Kielek** [comment]
@Yun-Ting, could you please update changelog and readme (if needed)?

File: src/OpenTelemetry.Instrumentation.Process/README.md [hunk 1]
```diff
@@ -61,17 +61,18 @@ allocated for the associated process.
...
+violation; including memory backed solely by RAM, pagefile-backed memory and
+file-backed memory.
 
 | Units | Instrument Type         | Value Type |
 |-------|-------------------------|------------|
 |  `By` | ObservableUpDownCounter | `Double`   |
 
 The API used to retrieve the value is:
 
-* [Process.WorkingSet64](https://learn.microsoft.com/dotnet/api/system.diagnostics.process.privatememorysize64):
```
Comment (Yun-Ting) [inline src/OpenTelemetry.Instrumentation.Process/README.md:73]: Note: this was an copy-paste error in the original document.

File: src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs [hunk 2]
```diff
@@ -51,10 +51,10 @@ public ProcessMetrics(ProcessInstrumentationOptions options)
             "process.memory.virtual",
             () =>
             {
-                return Diagnostics.Process.GetCurrentProcess().PrivateMemorySize64;
+                return Diagnostics.Process.GetCurrentProcess().VirtualMemorySize64;
             },
             unit: "By",
-            description: "The amount of virtual memory allocated for this process that cannot be shared with other processes.");
+            description: "All the address space this process can read from without trigerring an access violation.");
```
Comment (noahfalk) [inline src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs:57]:
```suggestion
            description: "The amount of committed virtual memory for this process.");
```
I think this is how most developers and tools would refer to it. I know it doesn't match the exact definition of committed memory Mark Russinovich gave in that blog post, but I think people will be more confused if we don't call it committed memory :)

File: src/OpenTelemetry.Instrumentation.Process/README.md [hunk 3]
```diff
@@ -61,17 +61,18 @@ allocated for the associated process.
 
 ### process.memory.virtual
 
-The amount of virtual memory allocated for this process
-that cannot be shared with other processes.
+All the address space this process can read from without trigerring an access
```
Comment (noahfalk) [inline src/OpenTelemetry.Instrumentation.Process/README.md:64]:
```suggestion
The amount of committed virtual memory for this process.
One way to think of this is all the address space this process can read from without trigerring an access
```

**Review by noahfalk — COMMENTED** [review]
Glad we are making this change 👍

File: src/OpenTelemetry.Instrumentation.Process/README.md [hunk 4]
```diff
@@ -61,17 +61,19 @@ allocated for the associated process.
 
 ### process.memory.virtual
 
-The amount of virtual memory allocated for this process
-that cannot be shared with other processes.
+The amount of committed virtual memory for this process.
+One way to think of this is all the address space this process can read from
+without trigerring an access violation; including memory backed solely by RAM,
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Process/README.md:66]:
```suggestion
without trigerring an access violation; this includes memory backed solely by RAM,
```

File: src/OpenTelemetry.Instrumentation.Process/README.md [hunk 5]
```diff
@@ -61,17 +61,19 @@ allocated for the associated process.
...
+pagefile-backed memory and file-backed memory.
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Process/README.md:67]: by a swapfile/pagefile and by other mapped files on disk.

Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Process/README.md:67]:
```suggestion
by a swapfile/pagefile and by other mapped files on disk.
```

**Review by utpilla — APPROVED** [review]

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Pull request merged**